packages/pharos/src/components/table/pharos-table.ts

```diff
@@ -0,0 +1,221 @@
+import { PharosElement } from '../base/pharos-element';
+import ScopedRegistryMixin from '../../utils/mixins/scoped-registry';
+import { html, nothing } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { state } from 'lit/decorators.js';
+import type { TemplateResult, CSSResultArray } from 'lit';
+import { property } from 'lit/decorators.js';
+import { tableStyles } from './pharos-table.css';
+import { PharosSelect } from '../select/pharos-select';
+import { PharosPagination } from '../pagination/pharos-pagination';
+
+export type ColumnSpecification = {
+  name: string;
+  field: string;
+};
+
+export type RowData = {
+  [key: string]: unknown;
+};
+
+/**
+ * Pharos table component.
+ *
+ * @tag pharos-table
+ *
+ * @fires pharos-table-prev-page - Fires when the table goes to previous page
+ * @fires pharos-table-next-page - Fires when the table goes to next page
+ *
+ */
+export class PharosTable extends ScopedRegistryMixin(PharosElement) {
+  static elementDefinitions = {
+    'pharos-select': PharosSelect,
+    'pharos-pagination': PharosPagination,
+  };
+
+  /**
+   * Column specification.
+   * @attr columns
+   * sample column:
+   * [
+   *  {
+   *    name: "ID"
+   *    field: "id"
+   *  },
+   *  {
+   *    name: "Message"
+   *    field: "message"
+   *  }
+   * ]
+   */
+  @property({ type: Array, reflect: true, attribute: 'columns' })
+  public columns: ColumnSpecification[] = [];
+
+  /**
+   * Row data.
+   * @attr row-data
+   * Example:
+   * [
+   *  {
+   *    id: 1,
+   *    message: "first one"
+   *  },
+   *  {
+   *    id: 2,
+   *    message: "second one"
+   *  }
+   * ]
+   */
+  @property({ type: Array, reflect: true, attribute: 'row-data' })
+  public rowData: RowData[] = [];
+
+  @property({ type: Boolean, reflect: true, attribute: 'hide-pagination' })
+  public hidePagination: boolean = true;
+
+  @property({ type: Number, reflect: true, attribute: 'total-results' })
+  public totalResults: number = 0;
+
+  @property({ type: Array, reflect: true, attribute: 'page-size-options' })
+  public pageSizeOptions: number[] = [50, 100];
+
+  @property({ type: String, reflect: true, attribute: 'caption' })
+  public caption: string = '';
+
+  @property({ type: String, reflect: true, attribute: 'hide-caption-visually' })
+  public hideCaptionVisually: boolean = false;
+
+  @state()
+  private _pageSize = 50;
+
+  @state()
+  private _currentPage = 1;
+
+  protected override firstUpdated(): void {
+    this._pageSize = this.hidePagination ? this.rowData.length : this.pageSizeOptions[0];
+    this.totalResults =
+      this.hidePagination || !this.totalResults ? this.rowData.length : this.totalResults;
```

Comment:
Why do we care about hidePagination here?

Reply:
So if the table doesn't have a pagination, consumer doesn't have to provide a `totalResults` attribute, we can automatically calculate it from the rowData.